neutron/test/constants_test.cpp

```diff
@@ -83,6 +99,35 @@ TEST_F(ConstantsTest, tof_to_energy) {
             L * L * Variable(constants::tof_to_energy_physical_constants));
 }
 
+TEST_F(ConstantsTest, tof_to_energy_transfer_fails) {
+  EXPECT_THROW(constants::tof_to_energy_transfer(dummy), std::runtime_error);
+  dummy.ei = 3.0 * units::meV;
+  dummy.ef = 3.0 * units::meV;
+  EXPECT_THROW(constants::tof_to_energy_transfer(dummy), std::runtime_error);
```

Comment:
Should we have specific exception types for Neutron scattering unit conversion?

Reply:
Not sure, might make them harder to catch?

Reply:
Would still be caught by current test case if derived from `std::runtime_error` but leave as is for now.